### DISCUSSION TITLE
How to best tackle realtime changes to databse

### DISCUSSION THREAD

**Discussion by Akxe** [prisma/prisma#2522] [Q&A] [answered]
I think there was some kind of subscriptions for Prisma 1, but I have not found any information about subscriptions for Prisma 2.

Is there anything to start from, when building a realtime (WebSocket) data server?

**Comment by ryands17** [comment, marked as answer]
Hey @Akxe :wave: 
Subscriptions are currently not supported in Prisma 2 and you can track the ongoing feature request [here](https://github.com/prisma/prisma/issues/298), but you can perform them in your business logic via apollo-server's `PubSub` module.

[Here](https://github.com/ryands17/nexus-auth/) is the repo if you need more info. It's a simple subscription implementation.

**Reply by Akxe** [reply]
@ryands17 Is there a plan to bring them back?

**Comment by hyochan** [comment]
@Akxe I've also created [example pr](https://github.com/dooboolab/ts-apollo-prisma/pull/3). I am currently working for a boilerplate to start new project with Prisma2 in [dooboolab/ts-apollo-prisma](https://github.com/dooboolab/ts-apollo-prisma).

**Reply by ryands17** [reply]
@hyochan and @Akxe :wave: 
There's a simple subscriptions [example](https://github.com/prisma/prisma-examples/tree/master/typescript/subscriptions-pubsub) added to the Prisma examples repo :smile:

**Reply by Akxe** [reply]
Thank you very much. Also thanks for showing how to use the new nexus. As I copied the old one from older examples before :/

---
Yet I wonder, why should the user spend so much time setting up every subscriútion? Since Prisma knows about every update I think some automated subscription would be awesome 😀.

Let say you requested 10 most recent posts. If someone posts a new post, You would be notified. If any of these post changes you would get differential changes made to it (along with all `id`s needed to pinpoint what changed in the model.

Thought this might be a better feature request for `nexus-prisma`.

**Comment by ludralph** [comment]
Hi @Akxe,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
